### DISCUSSION TITLE
Investigating war leadership mechanics

### DISCUSSION THREAD

**Discussion by moretrim** [moretrim/ccHFM#131] [Q&A]
After accepting a call to arms, a country can take over leadership of their side of the war… sometimes, but sometimes not. Known mechanics:

- Only the first accepted call to arms can take over. If original belligerent A calls B first and B takes over, any other ally C, D, etc. that accepts their own call from A will *not* take over even if they could take over a call from B. (“First” is a bit fuzzy here, it’s unclear if answering before any other country is what matters.)
- higher rank is not sufficient
- higher prestige is not sufficient
- higher military score is not sufficient

[Here is a provided save to test mechanics.](https://github.com/moretrim/ccHFM/files/7073091/PBC.cannot.take.leadership.zip) It is using [this particular prerelease version](https://github.com/moretrim/ccHFM/releases/tag/v1.1.0-prerelease.20210704). The setup is Peru-Bolivia receiving a call to arms from Germany to join in the Great War. Context for the war:

- original attacker is France, but attacker leadership has been taken over by Russia
- original defender is Germany

**Comment by rogerburks** [comment]
I can toss in something that may end up being a thinking box of sorts: Do we have a save that is a good example of many multi-party wars in which there can be variation in the war leader?

If so, then I suggest taking a look at the savefile and seeing if there is any number that sorts the tags in the war in the same way that the game lists them. Looking at an example, I think the focus could be only on `active_war`. Looking at the format (and I deleted the long list of battles), the game keeps track of when an attacker or defender is added, and it keeps track of the original attacker and original defender. Yet, we can assume that time may not be really the desired sorting factor here, but sorts them in the savefile.

If we find some number, hopefully in the savefile, that reliably sorts the tags as they are sorted on the war screen, then we have our sorting factor.

```
active_war=
{
	name="Dutch Colonial Conquest of Durrani Bas-Congo"
	history=
	{
		1935.10.10=
		{
			add_attacker="NET"
		}
		1935.10.10=
		{
			add_defender="DUR"
		}
		1935.10.10=
		{
			add_attacker="RUS"
		}
		1935.10.11=
		{
			add_attacker="BOR"
		}
		1935.10.11=
		{
			add_attacker="BEL"
		}

... (battles go here)

	original_attacker="NET"
	original_defender="DUR"
	original_wargoal=
	{
		state_province_id=1982
		casus_belli="place_in_the_sun"
		actor="NET"
		receiver="DUR"
	}
	action="1936.1.1"
}
```

**Idea 2:** It may also be desirable to create a number of wars through decision with allies called, and then to see what happens in a set of more or less controllable tests.
